server/server.js

```diff
@@ -88,9 +102,13 @@ async function handleValidRequest(button, numPresses, batteryLevel) {
     }
 
     let currentSession = await db.getUnrespondedSessionWithButtonId(button.button_id, client)
+    const currentTime = await db.getCurrentTime(client)
 
     if (batteryLevel !== undefined && batteryLevel >= 0 && batteryLevel <= 100) {
-      if (currentSession === null) {
+      if (
+        currentSession === null ||
+        (currentTime - currentSession.updatedAt >= SESSION_RESET_TIMEOUT && currentSession.state === ALERT_STATE.WAITING_FOR_REPLY)
+      ) {
```

Comment:
I was looking at this again and questioning whether it needed to check the state but based on `getUnrespondedSessionWithButtonId`'s check I think it does? It is late though so I'm not sure my brain is working right on that 😅

```
const query = 'SELECT * FROM sessions WHERE button_id = $1 AND state != $2 AND state != $3 AND state != $4'
const values = [buttonId, ALERT_STATE.WAITING_FOR_CATEGORY, ALERT_STATE.WAITING_FOR_DETAILS, ALERT_STATE.COMPLETED]
```

Reply:
You're right, `db.getUnrespondedSessionsWithButtonId` does check the state, so you don't need to check it here.

Also, whatever change you make to this `if`-statement also needs to be done on the one further down on line 126: `} else if (currentSession === null) {`. This happens in both these places because here is if we got a good battery level, and the other is if we didn't.

Reply:
ooh right that makes sense!